### ISSUE TITLE
Don't require Spacy for NLP model explanations

### ISSUE THREAD

**Issue by Hellisotherpeople** [SeldonIO/alibi#134]
The "ELI5" people who implement LIME in the form of TextExplainer offer a great blueprint on how to use the scikit learn tool-chain (which itself is extremely easy to work with) for allowing LIME explanations of any NLP classifier. 

I've succesfully used it for a multitude # of tasks including explaining word-embedding powered Keras NLP models and for multilabel model explanations (after a few hacks) 

I'd love to try out this anchor based method but the API expecting a Spacy object is a dealbreaker for me.

**Comment by jklaise** [comment]
Thanks for your interest in the library!

The way anchors work is different from LIME, in particular, in the text case for LIME features (e.g. words or n-grams) are randomly dropped from the test instance `x` to generate neighbouring instances to `x`. On the other hand, in anchors the features are not just dropped but are replaced by similar features. In the text case this involves sampling words by similarity to the token that is considered for modification, this requires having some corpus of vocabulary with pre-trained word similarities, the easiest off-the-shelf solution is to use `spacy` models for this.

May I ask what your use case is that having a spacy dependency would not work?

**Comment by Hellisotherpeople** [comment]
I think I was misunderstanding the role of Spacy. 

I thought that only Spacy models could be explained - not that Spacy is used for word vector similarity score

**Comment by jklaise** [comment]
Ah, I see, currently all our model explanation methods are black-box, so as long as you have access to a prediction function/endpoint, you're good to go!
